Userland/js.cpp

```diff
@@ -126,6 +127,25 @@ static void print_function(const JS::Object* function, HashTable<JS::Object*>&)
     printf("\033[34;1m[%s]\033[0m", function->class_name());
 }
 
+static void print_date(const JS::Object* date, HashTable<JS::Object*>&)
+{
+    fputs("Date(", stdout);
+    printf("\033[35;1m%d\033[0m", static_cast<const JS::Date*>(date)->datetime().year());
+    fputs(", ", stdout);
+    printf("\033[35;1m%d\033[0m", static_cast<const JS::Date*>(date)->datetime().month());
+    fputs(", ", stdout);
+    printf("\033[35;1m%d\033[0m", static_cast<const JS::Date*>(date)->datetime().day());
+    fputs(", ", stdout);
+    printf("\033[35;1m%d\033[0m", static_cast<const JS::Date*>(date)->datetime().hour());
+    fputs(", ", stdout);
+    printf("\033[35;1m%d\033[0m", static_cast<const JS::Date*>(date)->datetime().minute());
+    fputs(", ", stdout);
+    printf("\033[35;1m%d\033[0m", static_cast<const JS::Date*>(date)->datetime().second());
```

Comment:
Can we just call datetime().to_string("%format-string") instead of doing each component at a time? :)
Also, we don't need to set and unset the color every time; we can just output `\033[35;1m` at the start of the text, and `\033[0m` at the end.

Reply:
> Can we just call datetime().to_string("%format-string") instead of doing each component at a time? :)

Absolutely, I keep forgetting about that!

> Also, we don't need to set and unset the color every time; we can just output `\033[35;1m` at the start of the text, and `\033[0m` at the end.

My thinking behind that was to copy the printing style of arrays for example, having the numbers in colour and the commas and the "shell" around them in white - but if that's over-engineered I'll simplify it :)